### ISSUE TITLE
How can I add the static images into the cache?

### ISSUE THREAD

**Issue by squeezeboxhuf** [tatethurston/astrojs-service-worker#4]
I may be being a bit noob... I've added the astrojs-service-worker and it is working, in that it has created a service-worker.js for me and that has cached the static routes present in my project, however it has not added the images in my /public/images folder.

I guess I need to add some options to the serviceworker() integration but I can't tell from the docs what that might need to be (something to do with the globPatterns, I'm thinking...?)

Also, even with the routes cached (I can see them in the devtools application tab) I cannot navigate to them. On one device (PC with devtools network options set to offline) I get a nicely presented "You're offline" with my favicon logo. On my tablet, with the Wi-Fi turned off I get the default chrome error "No internet" dinosaur logo.

Confused...

Thanks. Hamish.

**Comment by tatethurston** [comment]
> however it has not added the images in my /public/images folder.

```dif
import { defineConfig } from "astro/config";
import serviceWorker from "astrojs-service-worker";

export default defineConfig({
  integrations: [serviceWorker({
+     workbox: {
+       include:  './public/images/*'
+     }
  })],
});
```

Could you try the above and let me know if it works for you?

> Also, even with the routes cached (I can see them in the devtools application tab) I cannot navigate to them. On one device (PC with devtools network options set to offline) I get a nicely presented "You're offline" with my favicon logo. On my tablet, with the Wi-Fi turned off I get the default chrome error "No internet" dinosaur logo.

1. Does your Tablet support service workers? What browser and version are you using?
2. Did you visit the site before turning the wifi off? The service worker has to install, which includes downloading all of the content it is set to cache, before it can serve content offline.

**Comment by squeezeboxhuf** [comment]
OK I got there, I have success!! I had to make  a couple of alterations... Firs in the astro.config.js I changed your "include" key to "globPattern", so it now looks like this:
```
import serviceWorker from "astrojs-service-worker";

// https://astro.build/config
import tailwind from "@astrojs/tailwind";

// https://astro.build/config
export default defineConfig({
  integrations: [tailwind(), serviceWorker({
    workbox: {
      globPatterns: ["**\/*.{js,css,html,png,jpg}"]
    }
  })]
});
```
... so it picks up the images. Then to make the offline links work I had to adjust my index page links so that they included the trailing slash. So the main home page lists a series of (32) linked dances, each one now goes to `/dance/0/`, `/dance/16/` etc, whereas before it was just `/dance/0`. I think that was upsetting the fetch from cache logic of the service-worker. 

I don't necessarily understand it, but it is successfully working for me as an installed offline app on my tablet! 👍 Thank you so much for the pointers, and the plug-in. 😀

**Comment by tatethurston** [comment]
Thanks for digging in here @squeezeboxhuf. I think it's reasonable for astrojs-service-worker to handle public folder without any configuration, so I'll look into that. 

I'm surprised by the trailing slash requirement for links -- I'll look into this. Do these links work without the trailing slash in production? If so, how are you deploying you application? Is it served out of a static server like AWS S3? Or something else?

Possibly related: https://docs.astro.build/en/reference/configuration-reference/#trailingslash

**Comment by squeezeboxhuf** [comment]
Yes it was OK in dev mode, and it works in production with a network connection. Just fails when I'm disconnected and working from the cache.  I'm developing using a docker VSCode instance on my Oracle cloud instance VM, the built version is being served by another docker container, I think using a base docker image from Nginx (I can't remember how I built the image now, but the container mounts the "dist" folder from the VSCode folder) but this project started as a Sveltekit development, then after struggling for a while with the Sveltekit Static adapter I realised if I was making a static site, maybe I should use a static dev platform! :)  I'll take a look at the trailingslash option, but for me for this project I just added the final slash to the links I'm generating with my array map on the index page and it works a treat online and offline.
